pipeline_dp/budget_accounting.py

```diff
@@ -1,20 +1,102 @@
+    constructed and the method compute_budgets was called.
+    """
+    _eps: float = None
+    _delta: float = None
+
+    @property
+    def eps(self):
+        """Parameter of (eps, delta)-differential privacy.
+
+        Raises:
+            AssertionError: The privacy budget is not calculated yet.
+        """
+        if self._eps is None:
+            raise AssertionError("Privacy budget is not calculated yet.")
+        return self._eps
+
+    @property
+    def delta(self):
+        """Parameter of (eps, delta)-differential privacy.
+
+        Raises:
+            AssertionError: The privacy budget is not calculated yet.
+        """
+        if self._delta is None:
+            raise AssertionError("Privacy budget is not calculated yet.")
+        return self._delta
+
+    def set_eps_delta(self, eps, delta):
+        self._eps = eps
+        self._delta = delta
+
+@dataclass
+class RequestedBudget:
+    """Manages the budget requested for an operation."""
+    budget: Budget
+    weight: float
+    use_eps: bool
+    use_delta: bool
 
 class BudgetAccountant:
-  """Manages privacy budget."""
+    """Manages the privacy budget."""
 
-  def __init__(self, epsilon, delta):
-    """Construct a BudgetAccountant
+    def __init__(self, epsilon, delta):
+        """Constructs a BudgetAccountant.
 
-    Args:
-      epsilon, delta: parameters of (epsilon, delta)-differential privacy.
-    """
-    if epsilon <= 0:
-      raise ValueError(f"Epsilon must be positive, not {epsilon}")
-    if delta < 0:
-      raise ValueError(f"Delta must non-negative, not {delta}")
-    self._eps = epsilon
-    self._delta = delta
-
-  # TODO: implement BudgetAccountant functionality.
+        Args:
+            epsilon, delta: Parameters of (epsilon, delta)-differential privacy.
+        """
+        if epsilon <= 0:
+            raise ValueError(f"Epsilon must be positive, not {epsilon}.")
+        if delta < 0:
+            raise ValueError(f"Delta must be non-negative, not {delta}.")
+
+        self._eps = epsilon
+        self._delta = delta
+        self._requested_budgets = []
+
+    def request_budget(self, weight, *, use_eps, use_delta):
+        """Requests a budget.
+
+        Args:
+            weight: The weight used to compute epsilon and delta for the budget.
+            use_eps: A boolean that is False when the operation doesn't need epsilon.
+            use_delta: A boolean that is False when the operation doesn't need delta.
+
+        Returns:
+            A "lazy" budget object that doesn't contain epsilon/delta until the
+            method compute_budgets is called.
+        """
+        budget = Budget()
+        requested_budget = RequestedBudget(budget, weight, use_eps, use_delta)
+        self._requested_budgets.append(requested_budget)
+        return budget
+
+    def compute_budgets(self):
+        """Computes the budgets after constructing the pipeline."""
```

Comment:
Please add to the function comment something like:

All previously requested Budget objects are updated with corresponding budget values.